components/brave_wallet/browser/zcash/rust/orchard_block_decoder_impl.h

```diff
@@ -6,18 +6,23 @@
 #ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_RUST_ORCHARD_BLOCK_DECODER_IMPL_H_
 #define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_RUST_ORCHARD_BLOCK_DECODER_IMPL_H_
 
+#include <memory>
 #include <vector>
 
 #include "brave/components/brave_wallet/browser/zcash/rust/orchard_block_decoder.h"
 
 namespace brave_wallet::orchard {
 
+class OrchardDecodedBlocksBundle;
+
 class OrchardBlockDecoderImpl : public OrchardBlockDecoder {
  public:
   ~OrchardBlockDecoderImpl() override;
 
-  std::optional<std::vector<::brave_wallet::OrchardNote>> ScanBlock(
-      const ::brave_wallet::zcash::mojom::CompactBlockPtr& block) override;
+  std::unique_ptr<OrchardDecodedBlocksBundle> ScanBlocks(
+      const ::brave_wallet::OrchardTreeState& tree_state,
+      const std::vector<::brave_wallet::zcash::mojom::CompactBlockPtr>& block)
```

Comment:
I think we should eventually fix namespaces somehow.
Like everything zcash related is in `brave_wallet::zcash`, everything orchard related is in  `brave_wallet::zcash::orchard` namespaces or something like that